### PR TITLE
Ejm merchant invoice update

### PR DESCRIPTION
**Pull request by EMielke76**

### CODE REVIEW

**Review by jaw772 — APPROVED** [review]

File: app/models/invoice.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@ def customer_name
   end
 
   def merchant_items(merchant)
+    # require "pry"; binding.pry
```
Comment (LelandCurtis) [inline app/models/invoice.rb]: let's delete this before we forget about it!

File: app/views/merchant_invoices/show.html.erb [hunk 2]
```diff
@@ -10,7 +10,11 @@
     <h3>Item: <%=invoice_item.item.name%></h3>
     <p>Quantity Ordered: <%=invoice_item.quantity%></p>
     <p>Unit Price: $<%=(invoice_item.unit_price.to_f/100)%></p>
-    <p>Status: <%=invoice_item.status%></p>
+      <div id="status-<%= invoice_item.item_id%>">
+        <%= form_with url: "/merchants/#{@merchant.id}/invoices/#{@invoice.id}/#{invoice_item.id}", method: :patch, local: true do |f| %>
```
Comment (LelandCurtis) [inline app/views/merchant_invoices/show.html.erb:14]: should we try refactoring this with the form_with :model? I wouldn't mind taking a stab at this to practice forms and advanced routing.

File: spec/features/merchants/invoices/show_spec.rb [hunk 3]
```diff
@@ -67,8 +59,8 @@
 
     visit "/merchants/#{merchant1.id}/invoices/#{invoice1.id}"
 
-    within("#invoice_#{item.id}") do
-      expect(page).to have_content("Status: shipped")
+    within("#status-#{item.id}") do
```
Comment (LelandCurtis) [inline spec/features/merchants/invoices/show_spec.rb:63]: perhaps change to status_ID rather than status-ID? super small thing, no big deal.

**Review by LelandCurtis — APPROVED** [review]
looks good, only minor comments and suggestions for refactoring.

File: app/models/invoice.rb (hunk 1 above)
Comment (EMielke76) [reply app/models/invoice.rb]: Removed!

File: app/views/merchant_invoices/show.html.erb (hunk 2 above)
Comment (EMielke76) [reply app/views/merchant_invoices/show.html.erb:14]: I had originally attempted the form_with model, but it wouldn't recognize the `invoice_item`, I ended up needing to interpolate the `invoice_item.status` later in the form to set the default value to what was coming in.

If you can make it work, I'm all for it :D

**Pull request merged**